src/modules/job-info/job_state.c

```diff
@@ -883,6 +883,116 @@ static flux_future_t *state_run_lookup (struct job_state_ctx *jsctx,
     return NULL;
 }
 
+static int eventlog_inactive_parse (struct info_ctx *ctx,
+                                    struct job *job,
+                                    const char *s)
+{
+    json_t *a = NULL;
+    size_t index;
+    json_t *value;
+    int rc = -1;
+
+    if (!(a = eventlog_decode (s))) {
+        flux_log (ctx->h, LOG_ERR,
+                  "%s: job %llu eventlog_decode: %s",
+                  __FUNCTION__, (unsigned long long)job->id, strerror (errno));
+        goto error;
+    }
+
+    json_array_foreach (a, index, value) {
+        const char *name = NULL;
+        json_t *context = NULL;
+
+        if (eventlog_entry_parse (value, NULL, &name, &context) < 0) {
+            flux_log (ctx->h, LOG_ERR,
+                      "%s: job %llu eventlog_entry_parse: %s",
+                      __FUNCTION__, (unsigned long long)job->id,
+                      strerror (errno));
+            goto error;
+        }
+
+        /* There is no need to check for "exception" events for the
+         * "success" attribute.  "success" is always false unless the
+         * job completes ("finish") without error.
+         */
+        if (!strcmp (name, "finish")) {
+            int status;
+            if (json_unpack (context, "{s:i}", "status", &status) < 0) {
+                flux_log (ctx->h, LOG_ERR,
+                          "%s: job %llu parse finish status",
+                          __FUNCTION__, (unsigned long long)job->id);
+                goto error;
+            }
+            if (!status)
+                job->success = true;
+        }
+    }
+
+    rc = 0;
+error:
+    json_decref (a);
+    return rc;
+}
+
+static void state_inactive_lookup_continuation (flux_future_t *f, void *arg)
+{
+    struct job *job = arg;
+    struct info_ctx *ctx = job->ctx;
+    struct state_transition *st;
+    const char *s;
+    void *handle;
+
+    if (flux_rpc_get_unpack (f, "{s:s}", "eventlog", &s) < 0) {
+        flux_log_error (ctx->h, "%s: error eventlog for %llu",
```

Comment:
Elsewhere we've been using the convention of `"%ju", (uintmax_t)id" for formatting job ID's.  Maybe use that instead?

Reply:
ahh, I have not mentally committed the `j` specifier to memory yet (I've only recently got `z` down).  I'll do it in a follow up cleanup PR since I use it all over the place.  And we apparently use it in `flux-job.c` all over the place too.